tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc

```diff
@@ -1815,9 +1927,50 @@ LogicalResult ConvertTFLeakyReluOp::matchAndRewrite(
       tf_leakyrelu_op.getResult().getType().dyn_cast<RankedTensorType>();
   if (!output_type) return failure();
 
-  // TODO: add lowering with MUL + SELECT
+  // Implement LeakyRelu as element-wise:
+  //   out = x > 0 ? x : alpha * x
+  //
+  // In TOSA ops:
+  //
+  //   const_zero = constant(0)
+  //   a1 = mul(x, alpha)
+  //   a2 = greater_equal(x, const_zero)
+  //   out = select(a2, x, a1)
+  //
+  // If alpha can be constrained to 0.0 <= alpha <= 1.0, then
+  // an alternative simpler lowering could be implemented with:
+  //
+  //   max(mul(x, alapha), x)
+  //
+  // But this alternative is not robust unless alpha meets those constraints.
+
+  FloatAttr tmpAttr = tf_leakyrelu_op.alphaAttr();
+  // There is disagreement between the MLIR .td defaults and TF
+  // documentation on 0.2 vs 0.3, but 0.2 will be used here.
+  double alpha = 0.2;
 
-  return failure();
+  if (tmpAttr) {
+    alpha = tmpAttr.getValueAsDouble();
+  }
+
+  Value const_zero = getTosaConstTensorSingleF32(rewriter, op, 0.0);
```

Comment:
I'm a bit concerned about the implicit assumption of f32 here (which is not an invariant of the tf op). What is the story for other data types (I believe if it happened, it would fail tosa verification, which is better than a crash, but still not great to be applying a pattern that produces a broken op).

Reply:
Maybe I should check if data type is f32 and return failure() is it's not now. Does that sound okay for now?

Reply:
Sounds fine.